comms/lighthouse/src/peers/archipelagoService.ts

```diff
@@ -27,7 +35,49 @@ export class ArchipelagoService {
     }
   }
 
+  get peersService() {
+    return this.peersServiceGetter()
+  }
+
   clearPeer(id: string) {
     this.controller.clearPeers(id)
   }
+
+  async onIslandUpdates(updates: IslandUpdates) {
+    const cachedIslands: Record<string, Island> = {}
+
+    const getIsland = async (id: string) => {
+      if (id in cachedIslands) return cachedIslands[id]
+
+      const island = await this.controller.getIsland(id)
+
+      if (island) {
+        cachedIslands[id] = island
+        return island
+      }
+    }
+
+    for (const id in updates) {
+      const update = updates[id]
+
+      const island = await getIsland(updates[id].islandId)
+      // This could be undefined for a short lived island, in the round trip between the worker & this service.
+      if (island) {
+        switch (update.action) {
```

Comment:
I'm just thinking out loud here, and it can definitely be left for another PR,  but would it make sense to group updates by island?

Because I'm seeing that we might be sending a lot of messages to the same peer to let them know about the updates on their island. And we could just send a message that is something like
```ts
export type ChangesInIsland = {
  type: PeerOutgoingMessageType.CHANGES_IN_ISLAND
  payload: {
    left: string[]
    joined: string[]
  }
}
```

Reply:
Yeah, I thought about this too... But in the end I decided to go for the simple implementation because I'm not really sure it'll be better.

Intuitively, it seems better (less messages over the network, right?). But I'm not sure we'll have much overhead, it being a websocket, and I'm also not sure the added latency will be worth the gains that we will have. It may also be better to forward the message directly to the socket, since it is native code and it should have better performance.

I think we should go with this simple implementation and once we have it working we can add performance tests and optimize.